src/ios.ts

```diff
@@ -132,11 +133,23 @@ export const iosApi = <RunTarget extends SupportedRunTarget<'ios'>>(
     target: { platform: 'ios', runTarget: options.runTarget },
     _internal: {
         ssh: async (...args) => {
+            let killProxyProcesss: (() => boolean) | undefined = undefined;
+
+            if (!options.targetOptions?.ip)
+                killProxyProcesss = await startUsbmuxProxy(22161, options.targetOptions?.port || 22);
+
             const ssh = await new NodeSSH().connect({
-                host: options.targetOptions!.ip,
+                host: options.targetOptions?.ip || '127.0.0.1',
+                port: options.targetOptions?.port ?? (options.targetOptions?.ip ? 22 : 22161),
```

Comment:
That's not the logic we want, is it? If `options.targetOptions?.ip` is not set, you start a usbmuxd proxy and interpret `options.targetOptions?.port` as the device port. You can't also use that here, then. You need `22161`.